### ISSUE TITLE
Demo of md-datepicker does not show ngmessages for Min / Max / Invalid

### ISSUE THREAD

**Issue by randallmeeker** [angular/material#4986]
Assuming the example is to show a message, it currently does not on the demo site (0.11.12) chrome 45.0.3454

<img width="271" alt="screen shot 2015-10-02 at 1 42 55 pm" src="https://cloud.githubusercontent.com/assets/2538537/10254825/d151ece4-690b-11e5-920f-7ed7c6363545.png">

**Comment by tjwoon** [comment]
Same problem here. The demo is from [Angular Material documentation](https://material.angularjs.org/0.11.2/#/demo/material.components.datepicker)

The line under the date turns red, but no textual messages.

**Comment by bastienJS** [comment]
With ng-messages inside a dialog:

http://plnkr.co/edit/B0xuv5pJyebju0SBb9gY?p=preview

Required message is shown with 0.11.2 in chrome!

**Comment by randallmeeker** [comment]
I just did your plnkr and got no messages shown and can not trigger the required validation. If I click in and set the date and erase it and click out, no required validation triggers. I can get an invalid validation, but still no message

Chrome 45.0.2454.101

<img width="281" alt="screen shot 2015-10-07 at 2 07 36 pm" src="https://cloud.githubusercontent.com/assets/2538537/10348118/e87f99b2-6cfc-11e5-9bb3-1a45116122b2.png">

**Comment by tjwoon** [comment]
@bastienJS I got red line but no message on Android Chrome Beta. Screenshot:

**Comment by randallmeeker** [comment]
I went back and tested the demo site and updated the title of this question accordingly. On the demo site required DOES show, however invalid / min / max do not show.

**Comment by jelbourn** [comment]
As of the latest release 1.0.0-rc2, this [appears to be fixed](https://material.angularjs.org/latest/demo/datepicker).

**Comment by randallmeeker** [comment]
Nope. Required shows, but not Min / Max , the title of this issue may be off because I noticed their is no message setup for invalid. 

<img width="770" alt="screen shot 2015-11-02 at 3 07 26 pm" src="https://cloud.githubusercontent.com/assets/2538537/10894307/7d62dc12-8173-11e5-8998-819194d659b8.png">

**Comment by jelbourn** [comment]
This is what I see on the demo site:
![image](https://cloud.githubusercontent.com/assets/838736/10895928/a0dfc586-816b-11e5-99b1-0a97a3d5235a.png)

@randallmeeker Which browser are you using?

**Comment by randallmeeker** [comment]
Chrome on Mac, just also tried on chrome for Android and can not see min/max either. What are you on?

**Comment by jelbourn** [comment]
Also Chrome on Mac. What are you doing to trigger the appearance of the error? Maybe there's a bug there.

**Comment by randallmeeker** [comment]
As you can see based on the previous post I'm making the year 2012. I'm also exiting the field by selecting the input above it.

**Comment by jelbourn** [comment]
Ahh, I see. This is actually a dupe of #5353; the ngModel $render isn't being invoked when the date changes via the keyboard for some reason.

**Comment by randallmeeker** [comment]
@jelbourn I'm just curious. how did you generate the onchange event without using the keyboard? The date selector does not allow selection of dates outside the min/max without use of the keyboard?

**Comment by jelbourn** [comment]
@randallmeeker Here's a summary of the issue:
* The `ng-change` expression will be executed whenever the ngModel value changes.
* The ngModel value only changes when the input (either selected from the calendar or typed) is valid (both a valid date and within min/max).  
* At first I thought the ngMessages weren't showing because the model change wasn't being picked up, but it was instead because the model (correctly) wasn't being updated. After #5510 is merged, everything should be working as intended.

**Comment by randallmeeker** [comment]
@jelbourn I perfectly understand. I was just curious how you bypassed the keyboard input to put in the date "9/23/2013" since keyboard input is not firing the onChange event and you can not select it via the calendar pop up. IE what were you doing differently to get your result than me?

**Comment by jelbourn** [comment]
@randallmeeker Ah- all the datepickers are bound to the same model, so I just changed one of the others on the page.

**Comment by randallmeeker** [comment]
@jpdesigndev ah, duh. thanks I totally missed that.

**Comment by latata** [comment]
@jelbourn I merged locally your pull request and it works, thanks. But what about changable mindate and maxdate attributes? For example - I have two datepickers and one (let's say datepicker A) needs date value greater than another (datepicker B). I pick a date in datepicker A and then I pick a date in datepicker B which is greater than value picked in datepicker A, so datepickewr A should return an error but it will not return an error till change in datepicker A.

**Comment by randallmeeker** [comment]
@latata I think in your situation you will have to manually trigger the onChange event?

**Comment by latata** [comment]
@randallmeeker you're right but I you think it will not come to the md-datepicker functionality? I won't be fixed and I should do this manually?

**Comment by randallmeeker** [comment]
@latata You're asking for the onChange event to be fired when the min/max values change. That is not probably something that needs to be fixed, but a feature you should ask for in a different issue.

**Comment by jelbourn** [comment]
@latata that should work now; minDate and maxDate are set with a two-way binding. If you're seeing a bug with that, could you file a new issue?

**Comment by askucher** [comment]
I didn't find validators in "model.$validators" of ng-model bound to ng-datepicker.
